### PR TITLE
feat: [select] support inputProps

### PR DESCRIPTION
**Pull request by pointhalo**
[中文模板 / Chinese Template](https://github.com/DouyinFE/semi-design/blob/main/.github/PULL_REQUEST_TEMPLATE.zh-CN.md)

- [ ] I have read and followed [Pull Request Guidelines](https://github.com/DouyinFE/semi-design/blob/main/CONTRIBUTING-en-US.md#pull-request-guidelines) of the contributing guide.

### What kind of change does this PR introduce? (check at least one)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update
 - [ ] Refactor
 - [ ] Test Case
 - [ ] TypeScript definition update
 - [ ] Document improve
 - [ ] CI/CD improve
 - [ ] Branch sync
 - [ ] Other, please describe:

#### 改动点
 - select renderInput增加 inputProps的消费
 - select api table移除一些较早的版本说明（v0.x的版本说明对开源用户基本无意义，可以移除掉）

### Changelog
🇨🇳 Chinese
- Select 新增 inputProps ，便于用户可实现一些特殊功能。例如传入 onCompositionEnd，onKeyDown事件监听等

---

🇺🇸 English
- Select newly added inputProps, which is convenient for users to realize some special functions. For example, incoming onCompositionEnd, onKeyDown event monitoring, etc.

### Checklist
- [x] Test or no need
- [x] Document or no need
- [ ] Changelog or no need

### CODE REVIEW

**Comment by pointhalo** [comment]
filterInputProps Or inputProps?

**Comment by shijiatongxue** [comment]
codesandbox 编译有报错

image

**Comment by pointhalo** [comment]
codesandbox的报错是scss版本的问题。可以忽略。daiqiang已经在main里改过了。

File: packages/semi-ui/select/index.tsx [hunk 1]
```diff
@@ -115,6 +115,7 @@ export type SelectProps = {
     suffix?: React.ReactNode;
     prefix?: React.ReactNode;
     insetLabel?: React.ReactNode;
+    inputProps?: Record<string, any>;
```
Comment (shijiatongxue) [inline packages/semi-ui/select/index.tsx:118]: 这里可以用 InputProps，Exclude 掉不需要属性或不建议传的属性？

Comment (pointhalo) [reply packages/semi-ui/select/index.tsx]: done，再re一下

**Pull request merged**